web/frontend/src/layout/App.tsx

```diff
@@ -27,15 +27,20 @@ import { AuthContext } from '..';
 import Logged from 'pages/session/Logged';
 import Flash from './Flash';
 import NotFound from './NotFound';
+import { ROLE } from 'types/userRole';
 
 const App = () => {
-  const RequireAuth = ({ children }) => {
+  const RequireAuth = ({ children, roles }: { children: any; roles?: string[] }): any => {
     let location = useLocation();
 
     const authCtx = useContext(AuthContext);
 
     if (!authCtx.isLogged) {
       return <Navigate to="/login" state={{ from: location }} replace />;
+    } else {
+      if (roles && !roles.includes(authCtx.role)) {
+        return <Navigate to="/notfound" state={{ from: location }} replace />;
```

Comment:
you should return to home, with a flash message saying that the user is not authorized.